internal/bminventory/inventory.go

```diff
@@ -2987,6 +2987,31 @@ func (b *bareMetalInventory) processDiskSpeedCheckResponse(ctx context.Context,
 	return b.hostApi.SetDiskSpeed(ctx, h, diskPerfCheckResponse.Path, diskPerfCheckResponse.IoSyncDuration, exitCode, nil)
 }
 
+func (b *bareMetalInventory) updateDomainNameResolutionResponse(ctx context.Context, host *models.Host, domainResolutionResponseReporter string) error {
+	var (
+		err                      error
+		domainResolutionResponse models.DomainResolutionResponse
+	)
+	log := logutil.FromContext(ctx, b.log)
+	log.Debugf("The address for domain name resolotion on host %s are: %s", host.ID.String(), domainResolutionResponse)
+
+	if err = json.Unmarshal([]byte(domainResolutionResponseReporter), &domainResolutionResponse); err != nil {
+		log.WithError(err).Warnf("Json unmarshal domain name resolution of host %s", host.ID.String())
+		return err
+	}
+	if len(domainResolutionResponse.Resolutions) == 0 {
+		err = errors.Errorf("Domain name resolution of host %s is empty", host.ID.String())
+		log.WithError(err).Warn("Check host DNS configuration")
+		return err
+	}
+	if err = b.db.Model(&common.Host{}).Where("id = ? and cluster_id = ?", host.ID.String(),
```

Comment:
Updates should be part of hostApi, not in inventory.go

Reply:
Other steps also update in `inventory.go`

Reply:
The fact that something else has component isolation, does not mean that we should continue with it

Reply:
Didn't say we should, just thought I should mention that this is already the status-quo so it's worth reconsidering the review comment

Reply:
Most of the responses (and that's the way we follow), that the updates are part of API

Reply:
Moved the saving to the hostAPI